hw/ip/ascon/data/ascon.hjson

```diff
@@ -0,0 +1,696 @@
+      hwqe:     "true",
+      fields: [
+        { bits: "31:0", name: "nonce", desc: "Input Nonce Share 0" }
+      ],
+      }
+    },
+    { multireg: {
+      name: "NONCE_SHARE1",
+      desc: '''
+        Input Nonce Register Share 1.
+        The actual data corresponds to Nonce Input Registers Share 0 XORed with Nonce Input Registers Share 1.
+        If SW does not want to provide the Nonce masked, it can simply set one share to all zeros and provide the unmasked nonce in the other share.
+        All nonce registers (Share0 and Share 1) must be written each time a new message is processed.
+        Ascon requires the nonce to be unique for each message. However there are no hardware checks to enforce this.
+        The order in which the registers are updated does not matter.
+        If the user fails to update the register the NO_NONCE flag is set in the ERROR Register. Can only be updated when the Ascon unit is idle.
+        If the Ascon unit is non-idle, writes to these registers are ignored. Upon reset, these registers are cleared with pseudo-random data.
+      '''
+      count: "NumRegsNonce",
+      cname: "NONCE_SHARE1",
+      swaccess: "wo",
+      hwaccess: "hrw",
+      hwext:    "true"
+      hwqe:     "true",
+      fields: [
+        { bits: "31:0", name: "nonce", desc: "Input Nonce Share 1" }
+      ],
+      }
+    },
+
+
+##############################################################################
+# input message registers
+    { multireg: {
+      name: "DATA_IN_SHARE0",
+      desc: '''
+        Input Data Register 0.
+        The actual data corresponds to Data Input Registers Share 0 XORed with Data Input Registers Share 1.
+        This register holds share0 of one input block.
+        This is either the associated data, plaintext or the ciphertext, depending on the mode.
+        All registers must be written each time a new message is processed.
+        For Ascon 128 the upper 64 bit can be set to any value. For partial blocks the unused bytes can be set to any value.
+        The order in which the registers are updated does not matter.
+      '''
+      count: "NumRegsData",
+      cname: "MSG_IN_SHARE0",
+      swaccess: "wo",
+      hwaccess: "hrw",
+      hwext:    "true",
+      hwqe:     "true",
+      fields: [
+        { bits: "31:0", name: "msg_in", desc: "Input Data Share 0" }
+      ],
+      }
+    },
+    { multireg: {
+      name: "DATA_IN_SHARE1",
+      desc: '''
+        Data Input Register Share 1
+        The actual message corresponds to Data Input Registers Share 0 XORed with Data Input Registers Share 1.
+        This register holds share1 of one data input block.
+        This is either the associated data, plaintext or the ciphertext, depending on the mode.
+        If CTRL_SHADOWED.masked_{ad,msg}_input = 0, share 1 must be set to all zeros before the first block of
+        the associated data / message is written to share 0. This basically disables input masking.
```

Comment:
Nit: 1 sentence per line.

Reply:
I would change this to:
```
 If CTRL_SHADOWED.masked_{ad,msg}_input = 1, all registers must be written each time a block of associated data or message is processed.
 If CTRL_SHADOWED.masked_{ad,msg}_input = 0, the write operation tracking for these registers is disabled.
Software should set Share 1 to all zero before the first block of associated data / message is written to Share 0.
This basically disables input masking.
```
Because the hardware won't check if the values are indeed 0, nor will it check whether they have been written. It's up to software to do the right thing.